File: Servidor/EventManagement/src/main/java/com/lausebscode/modelo/Organizador.java
package com.lausebscode.modelo;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.PastOrPresent;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class Organizador {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @NotBlank(message = "El nombre no puede estar vacío")
    private String nombre;

    @NotNull
    @PositiveOrZero(message = "El presupuesto no puede ser negativo")
    private double presupuesto;

    @NotNull
    @PastOrPresent(message = "La fecha de fundación no puede ser futura")
    private LocalDateTime fundacion;

    @NotBlank(message = "El CEO no puede estar vacío")
    private String ceo;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "feria_gastro_id")
    @JsonBackReference
    private FeriaGastro feriaGastro;
}

File: Cliente/dotnet-eventmng-p3/dotnet-eventmng-p3/Modelo/Organizador.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace dotnet_eventmng_p3.Modelo
{
    public class Organizador
    {
        public int id { get; set; }
        public string nombre { get; set; }
        public double presupuesto { get; set; }
        public DateTime fundacion { get; set; }
        public string ceo { get; set; }

        public Organizador(int id, string nombre, double presupuesto, DateTime fundacion, string ceo)
        {
            this.id = id;
            this.nombre = nombre;
            this.presupuesto = presupuesto;
            this.fundacion = fundacion;
            this.ceo = ceo;
        }
    }
}
